nvflare/apis/client.py

```diff
@@ -67,3 +73,52 @@ def set_is_leaf(self, value: bool):
 
     def get_is_leaf(self):
         return self.get_prop(ClientPropKey.IS_LEAF)
+
+    def to_dict(self) -> dict:
+        """Convert the Client object to a dict representation.
+        This dict could be used included into a job's metadata.
+
+        Returns: dict that contains essential info of the client.
+
+        Note that the client's token is not included in the result since it is authentication data.
+
+        """
+        r = {ClientDictKey.NAME: self.name}
+
+        fqcn = self.get_fqcn()
+        if fqcn != self.name:
+            r[ClientDictKey.FQCN] = fqcn
+
+        fqsn = self.get_fqsn()
+        if fqsn != self.name:
+            r[ClientDictKey.FQSN] = fqsn
+
+        is_leaf = self.get_is_leaf()
+        if not is_leaf:
+            r[ClientDictKey.IS_LEAF] = False
+
+        return r
+
+
+def from_dict(d: dict) -> Client:
+    """Create a Client object from the data in the specified dict.
+
+    Args:
+        d: the dict that contains Client data. This dict should be the result of to_dict() of a Client object.
+
+    Returns: a Client object
+
+    """
+    name = d.get(ClientDictKey.NAME)
+    c = Client(name=name, token="")
+
+    fqcn = d.get(ClientDictKey.FQCN, name)
+    c.set_fqcn(fqcn)
```

Comment:
Should check for None here.

Reply:
fqcn is default to "name" if no FQCN in the dict.